### DISCUSSION TITLE
Excluding some imports from automatic organization

### DISCUSSION THREAD

**Discussion by arrufat** [zigtools/zls#2093] [Q&A] [answered]
Hi, is there a way to prevent some imports from being reorganized at the top of the file?

Something like `zig fmt: off` but for `organize @import`?

**Comment by techatrix** [comment, marked as answer]
This isn't currently possible. May I ask what you would intent to do with such a feature?

**Reply by arrufat** [reply]
I have some functions that I declare with `callConv(.C)` in a file, then, I `@import` them in the main file and `export` them from there. They get scattered across my other imports, but I'd like to have them grouped.

Something like this:

`main.zig`
```zig
pub export const print_version = @import("api.zig").print_version;
```

Because in the `api.zig` I have functions that are common to several libraries:

`api.zig`
```zig
pub fn print_version() callconv(.C) void {
    std.log.info("{s}", .{@import("build_options").version});
}
```

This way I can define all my API functions in the same place, but then, only export the ones I want to use for a particular library.

It's a weird usecase, I know...
